### PR TITLE
Add guards for geo index columns which are complex expressions.

### PR DESCRIPTION
**Pull request by BillWhite**
We can't actually create geospatial indices with columns which are more
complicated expressions than simple column references.

https://issues.voltdb.com/browse/ENG-9695

### CODE REVIEW

File: tests/frontend/org/voltdb/compiler/TestDDLCompiler.java [hunk 1]
```diff
@@ -806,6 +806,32 @@ void testFailure(String ddl) {
         tester.testSuccess("create table an_unquoted_table (an_unquoted_column integer)");
     }
 
+    public void testIndexExpresions() throws Exception {
```
Comment (paulmartel) [inline tests/frontend/org/voltdb/compiler/TestDDLCompiler.java]: name spelling

File: src/frontend/org/voltdb/compiler/DDLCompiler.java [hunk 2]
```diff
@@ -2066,10 +2066,16 @@ else if (subNode.name.equals("predicate")) {
...
-                    if (has_geo_col && exprs.size() > 1) {
-                        String emsg = "Cannot create index \""+ name + "\" because " +
-                                colType.getName() + " values must be the only component of an index key.";
-                        throw compiler.new VoltCompilerException(emsg);
+                    if (has_geo_col) {
+                        if (exprs.size() > 1) {
+                            String emsg = "Cannot create index \""+ name + "\" because " +
+                                        colType.getName() + " values must be the only component of an index key.";
+                            throw compiler.new VoltCompilerException(emsg);
+                        } else if (!(expression instanceof TupleValueExpression)) {
```
Comment (paulmartel) [inline src/frontend/org/voltdb/compiler/DDLCompiler.java]: I don't see the point of an else after a throw. It just seems like noise to me.

**Comment by paulmartel** [comment]
Looks correct.

**Comment by BillWhite** [comment]
Jenkins is blue.  I'm going to squash this separately.

**Pull request closed without merging**